fedimint-core/src/runtime.rs

```diff
@@ -163,3 +163,77 @@ mod r#impl {
         }
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use tokio::sync::oneshot;
+
+    use super::*;
+
+    #[tokio::test]
+    async fn test_spawn_and_await() {
+        let name = "test_spawn_and_await";
+        let (tx, _rx) = oneshot::channel();
+
+        let future = async move {
+            // Simulating some work
+            tokio::time::sleep(Duration::from_millis(1)).await;
+            let _ = tx.send("done");
```

Comment:
I don't understand what this test is trying to test.

But probably the `future` should be only `rx.recv()`;

Then `spawn` that future as a task, then  `tx.send()` to unblock it, then `await` on the result.